common/src/main/java/com/hortonworks/streamline/common/util/ReflectionHelper.java

```diff
@@ -47,7 +47,24 @@
 
     public static <T> T invokeSetter(String propertyName, Object object, Object valueToSet) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
         String methodName = "set" + StringUtils.capitalize(propertyName);
-        Method method = object.getClass().getMethod(methodName, valueToSet.getClass());
+        Method method = null;
+        try {
+            method = object.getClass().getMethod(methodName, valueToSet.getClass());
+        } catch (NoSuchMethodException ex) {
+            // try setters that accept super types
+            Method[] methods = object.getClass().getMethods();
+            for (int i = 0; i < methods.length; i++) {
+                if (methods[i].getName().equals(methodName) && methods[i].getParameterCount() == 1) {
+                    if (methods[i].getParameters()[0].getType().isAssignableFrom(valueToSet.getClass())) {
```

Comment:
It will apply the first match. Typically there should be only one match. Overloading methods with subtypes is **bad** and should not be done since Java resolves method overload based on the declared type and not the actual type of the argument.